prebid/native-implementation.md

```diff
@@ -191,158 +268,428 @@ pbjs.addAdUnits({
-1. If the template was defined in the AdServer, the body of the iframe will be replaced. Otherwise, if the template was defined in the AdUnit or external JavaScript, the body of the iframe will be appended with resolved creative HTML.
-1. If the AdServer supports it, CSS styles can be defined in the iframe head. Otherwise, they must be defined in the body of the template, for example with <style> tags.
-1. Note that native iframes will be resized to the height of the creative after render.
+1. The native assets are stored in the Prebid.js memory space as the original OpenRTB response.
+2. When the `window.pbNativeTag.renderNativeAd()` function is called, an HTML5 postmessage is made. Prebid.js is listening for this message, and responds with the bidder's OpenRTB assets.
+3. If the template was defined in the AdServer, the body of the iframe will be replaced. Otherwise, if the template was defined in the AdUnit or external JavaScript, the body of the iframe will be appended with resolved creative HTML.
+4. If the AdServer supports it, CSS styles can be defined in the iframe head. Otherwise, they must be defined in the body of the template, for example with <style> tags.
+5. Note that native iframes will be resized to the height of the creative after render.
+
+## 6. Event tracking
+Native ads support 3 different types of event tracking:
+- impression tracking
+- click tracking
+- viewability tracking
+
+### 6.1. Impression Tracking
+Prebid supports both methods (`img` and `js`) for impression tracking. Publishers just need to specify which method is support for which adUnit. This is configured in `adUnit.mediaTypes.native.ortb.eventtrackers`. For more details take a look at OpenRTB specs.  
+
+How impression tracking works step by step:
+- When the `window.pbNativeTag.renderNativeAd()` is called, the Prebid Universal Creative will request OpenRTB response from Prebid.js (via postmessage).
+- Prebid.js responds with OpenRTB response that contains `eventtrackers`
+- When the Prebid Universal Creative renders the native ad, it will gather all impression trackers from the response and for each tracker it will:
+  - if tracker is `img` method, Prebid Universal Creative will fire pixel provided in the `url` property
+  - if tracker is `js` method, Prebid Universal Creative will load the script in the same iframe where native ad is rendered
+
+The publisher doesn't need to implement anything for impression tracking to work, all that is needed is proper configuration of OpenRTB request on adUnit level. Prebid.js + Prebid Universal Creative will automatically take care of the impression tracking.
+
+### 6.2. Click Tracking
+According to the OpenRTB spec, click trackers can be found in `link` objects. There is a 'master' `link` object for the entire native ad and also each asset in the response can have its own separate `link` object.
```

Comment:
I'm all about linking stuff. "OpenRTB spec" is always a bit vague as there are so many.
In this case you are (probably) referencing the OpenRTB Native Ads 1.2 Spec :wink: 

```suggestion
According to the [OpenRTB spec](https://www.iab.com/wp-content/uploads/2018/03/OpenRTB-Native-Ads-Specification-Final-1.2.pdf], click trackers can be found in `link` objects. There is a 'master' `link` object for the entire native ad and also each asset in the response can have its own separate `link` object.
```

Reply:
fixed